core/src/main/kotlin/com/strumenta/kolasu/parsing/Parsing.kt

```diff
@@ -36,12 +36,27 @@ open class CodeProcessingResult<D>(
     }
 }
 
+data class TokenCategory(val type: String) {
+    companion object {
+        val COMMENT = TokenCategory("Comment")
+        val KEYWORD = TokenCategory("Keyword")
+        val NUMERIC_LITERAL = TokenCategory("Numeric literal")
+        val STRING_LITERAL = TokenCategory("String literal")
+        val PLAIN_TEXT = TokenCategory("Plain text")
+    }
+}
+
+open class KolasuToken(open val category: TokenCategory, open val position: Position)
+
+data class KolasuANTLRToken(override val category: TokenCategory, val token: Token) :
```

Comment:
Is this an adapter from the ANTLR Token to the KolasuToken? Maybe we could add a comment to specify that